aws_lambda_powertools/event_handler/openapi/encoders.py

```diff
@@ -69,88 +70,94 @@ def jsonable_encoder(  # noqa: PLR0911
     if exclude is not None and not isinstance(exclude, (set, dict)):
         exclude = set(exclude)
 
-    # Pydantic models
-    if isinstance(obj, BaseModel):
-        return _dump_base_model(
-            obj=obj,
-            include=include,
-            exclude=exclude,
-            by_alias=by_alias,
-            exclude_unset=exclude_unset,
-            exclude_none=exclude_none,
-            exclude_defaults=exclude_defaults,
-        )
+    try:
+        # Pydantic models
+        if isinstance(obj, BaseModel):
+            return _dump_base_model(
+                obj=obj,
+                include=include,
+                exclude=exclude,
+                by_alias=by_alias,
+                exclude_unset=exclude_unset,
+                exclude_none=exclude_none,
+                exclude_defaults=exclude_defaults,
+            )
 
-    # Dataclasses
-    if dataclasses.is_dataclass(obj):
-        obj_dict = dataclasses.asdict(obj)
-        return jsonable_encoder(
-            obj_dict,
-            include=include,
-            exclude=exclude,
-            by_alias=by_alias,
-            exclude_unset=exclude_unset,
-            exclude_defaults=exclude_defaults,
-            exclude_none=exclude_none,
-        )
+        # Dataclasses
+        if dataclasses.is_dataclass(obj):
+            obj_dict = dataclasses.asdict(obj)
+            return jsonable_encoder(
+                obj_dict,
+                include=include,
+                exclude=exclude,
+                by_alias=by_alias,
+                exclude_unset=exclude_unset,
+                exclude_defaults=exclude_defaults,
+                exclude_none=exclude_none,
+            )
 
-    # Enums
-    if isinstance(obj, Enum):
-        return obj.value
+        # Enums
+        if isinstance(obj, Enum):
+            return obj.value
 
-    # Paths
-    if isinstance(obj, PurePath):
-        return str(obj)
+        # Paths
+        if isinstance(obj, PurePath):
+            return str(obj)
 
-    # Scalars
-    if isinstance(obj, (str, int, float, type(None))):
-        return obj
+        # Scalars
+        if isinstance(obj, (str, int, float, type(None))):
```

Comment:
suggestion: take a look at `singledispatch` to create a dispatcher to make this more maintainable, as it grows with the different types we're handling it.

Reply:
(seeing on laptop now) - it sounds like those were already there, so disregard it. 

it's easier to have an issue to refactor these parts

Reply:
I'll open an issue to refactor that.